### PR TITLE
feat: Output contract-id to .out folder with forc build

### LINKED ISSUE #2930: Introduce a way for `forc-build` to output deployment information
Currently the information related to a specific deployment is only `print`ed and we do not have a non-volatile way of storing this information. We may want to output something like a deployment file which is output in a predetermined structure so that later on workspace members can be aware of some deployment information (like the contract id) of other members so that we can send that information in between them.

### PR DESCRIPTION
**Pull request by kayagokalp**
closes #2930.

### CODE REVIEW

**Review by mohammadfawaz — APPROVED** [review]
This all seems reasonable but I wonder if we can reuse this code in https://github.com/FuelLabs/sway/blob/4740fb2f9b9c4c4d6e8ba2110831604b40ccc645/forc-plugins/forc-client/src/ops/deploy/op.rs#L148 and wherever else we compute the contract ID in that file.

**Comment by mitchmindtree** [comment]
@mohammadfawaz good spot! The plan is to eventually have `forc-client` use this :+1: https://github.com/FuelLabs/sway/issues/2929#issuecomment-1267824974

File: forc-pkg/src/pkg.rs [hunk 1]
```diff
@@ -2118,12 +2118,25 @@ pub fn build_with_options(build_options: BuildOptions) -> Result<Compiled> {
...
             let res = if minify_json_storage_slots {
-                serde_json::to_writer(&file, &compiled.storage_slots)
+                serde_json::to_writer(&storage_slots_file, &compiled.storage_slots)
             } else {
-                serde_json::to_writer_pretty(&file, &compiled.storage_slots)
+                serde_json::to_writer_pretty(&storage_slots_file, &compiled.storage_slots)
             };
+            // Construct the contract ID
+            let contract = Contract::from(compiled.bytecode.clone());
+            let salt = fuel_tx::Salt::new([0; 32]);
```
Comment (mitchmindtree) [inline forc-pkg/src/pkg.rs:2129]: Maybe we can open an issue for specifying the salt via flag?

**Review by mitchmindtree — APPROVED** [review]
:ok_hand:

**Pull request merged**